lib/pbench/test/functional/server/test_put.py

```diff
@@ -120,22 +120,28 @@ def check_indexed(server_client: PbenchServerClient, datasets):
             for dataset in datasets:
                 print(f"\t... on {dataset.metadata['server.tarball-path']}")
                 metadata = server_client.get_metadata(
-                    dataset.resource_id, ["dataset.state", "server.status"]
+                    dataset.resource_id, ["dataset.operations"]
                 )
-                state = metadata["dataset.state"]
-                status = metadata["server.status"]
-                stats = set(status.keys()) if status else set()
-                if state == "Indexed" and {"unpack", "index"} <= stats:
-                    # Don't wait for backup, and don't fail if we haven't as
-                    # it's completely independent from unpack/index; but if we
-                    # have backed up, check that the status was successful.
-                    if "backup" in stats:
-                        assert status["backup"] == "ok"
-                    assert status["unpack"] == "ok"
-                    assert status["index"] == "ok"
+                operations = metadata["dataset.operations"]
+                if "INDEX" in operations and operations["INDEX"]["state"] == "OK":
+                    assert operations["UPLOAD"]["state"] == "OK"
+                    assert operations["UNPACK"]["state"] == "OK"
+                    assert operations["INDEX"]["state"] == "OK"
+
+                    # Backup is asynchronous: it's OK if it hasn't completed
+                    # yet, but must be at least in READY state.
+                    assert operations["BACKUP"]["state"] in ("OK", "READY")
```

Comment:
It cannot be `"WORKING"`?  (Presumably it should not be `"FAILED"`....)

Reply:
I debated how widely I wanted to use `WORKING`. In the end, I (lazily) used it only for publish and delete rather than doing the extra update at the beginning of other operations. I put off thinking about this semantic issue in solving the real problem and then never really decided to worry about it.

Reply:
So...I should just resolve this?